.github/workflows/verify-pr-commit.yml

```diff
@@ -734,3 +730,89 @@ jobs:
+      - name: Run Benchmarks?
+        id: run-benchmarks
+        shell: bash
+        run: |
+          commit_message=$(git show -s --format=%s)
+          echo "commit_message: $commit_message"
+          regex='\[run-benchmarks (.*)\]$'
+          if [[ $commit_message =~ $regex ]]; then
+            echo "Need to run benchmarks."
+            pallets="${BASH_REMATCH[1]}"
+            echo "Pallets: $pallets"
+            echo "RUN_BENCHMARKS=true" >> $GITHUB_ENV
+            echo "pallets=$pallets" >> $GITHUB_OUTPUT
+          else
+            echo "Do not need to run benchmarks."
+            echo "RUN_BENCHMARKS=false" >> $GITHUB_ENV
+            exit 0
+          fi
+      - name: Set PR Commit Status to Pending
+        if: env.RUN_BENCHMARKS == 'true'
+        uses: ouzi-dev/commit-status-updater@219d3f932547cad092e384c7a36bf4d963739c35
+        with:
+          name: "GithubActions - Run Benchmarks"
+          status: "pending"
+          addHoldComment: "true"
+          pendingComment: ":hourglass_flowing_sand: Running benchmarks and calculating weights..."
+          failComment: "ERROR: Failed to set PR commit to pending status!"
+      - name: Install Required Packages
+        if: env.RUN_BENCHMARKS == 'true'
+        run: |
+          sudo apt-get update
+          sudo apt install -y protobuf-compiler libclang-dev
+      - name: Install Rust Toolchain
+        if: env.RUN_BENCHMARKS == 'true'
+        uses: dtolnay/rust-toolchain@e12eda571dc9a5ee5d58eecf4738ec291c66f295
+        with:
+          toolchain: stable
+      - name: Run Benchmarks
+        if: env.RUN_BENCHMARKS == 'true'
+        env:
+          BENCHMARK_SCRIPT: ./scripts/run_benchmark.sh
+        run: |
+          set -x
+          pallets_str="${{steps.run-benchmarks.outputs.pallets}}"
+          echo "Pallets: $pallets_str"
+          if [ pallets_str = 'all' ]; then
+            echo "Running benchmarks for all pallets..."
+            make run-benchmarks
+          else
+            IFS=',' read -r -a pallets <<< "$pallets_str"
+            for pallet in "${pallets[@]}"
+            do
+              echo "Running benchmarks for $pallet pallet..."
+              $BENCHMARK_SCRIPT -p $pallet
+              echo "Finished benchmarks for $pallet pallet."
+            done
+          fi
+      - name: Print Updated Weights
+        run: |
+          git status
+          git diff
+      - name: Commit Updated Weights
+        if: steps.git-check.outputs.repo_modified == 'true'
+        run: |
+          git config --global user.name 'Frequency CI'
+          git config --global user.email 'do-not-reply@users.noreply.github.com'
+          git commit -am "Update weights in PR #${{github.event.pull_request.number}}"
+          git push origin
+      - name: Set PR Commit Status to Success
+        if: env.RUN_BENCHMARKS == 'true'
+        uses: ouzi-dev/commit-status-updater@219d3f932547cad092e384c7a36bf4d963739c35
```

Comment:
Do we need to set `ignoreForks: "false"`? We don't really want to run benchmarks without it being from a core dev however. Thoughts?